### PR TITLE
Introduce default tenant mapping handler component

### PR DESCRIPTION
**Pull request by nyordanoff**
**Description**
This PR introduces a new component which is responsible for providing a default tenant mapping implementation. It always returns `READY` in ASYNC way without any business logic.

Changes proposed in this pull request:
- skeleton of the new component
- handler implementation of the new component
- authz check when the new component calls the status API with the CMP certificate
- e2e test

**Related issue(s)**

- https://github.com/kyma-project/test-infra/pull/8912

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/default-tenant-mapping-handler/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    app.kubernetes.io/name: {{ template "name" . }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "fullname" . }}
+    namespace: {{ $.Release.Namespace }}
+roleRef:
+  kind: Role
+  name: director-external-client-certificate
```
Comment (ognyvrac) [inline chart/compass/charts/default-tenant-mapping-handler/templates/rbac.yaml:27]: `director-{{ .Values.global.externalCertConfiguration.secrets.externalClientCertSecret.name }}` - that is how the role is created, less hard-coded

File: chart/compass/charts/default-tenant-mapping-handler/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {{- include "defaultTenantMappingHandler.labels" . | nindent 4 }}
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "fullname" . }}
+  namespace: {{ $.Release.Namespace }}
+  labels:
+    app: {{ $.Chart.Name }}
+    release: {{ $.Release.Name }}
```
Comment (ognyvrac) [inline chart/compass/charts/default-tenant-mapping-handler/templates/rbac.yaml:16]: everything below the `app` label should be covered with `{{- include "defaultTenantMappingHandler.labels" . | nindent 4 }}`

**Review by emmakarova — APPROVED** [review]

**Review by StanislavStefanov — APPROVED** [review]

**Pull request merged**